langchain/src/retrievers/recency_ranked.ts

```diff
@@ -0,0 +1,74 @@
+import { BaseRetriever } from "@langchain/core/retrievers";
+import { VectorStoreInterface } from "@langchain/core/vectorstores";
+import { Document } from "@langchain/core/documents";
+
+export interface RecencyRankedRetrieverConfig {
+  vectorStore: VectorStoreInterface;
+  k: number;
+  top_k?: number;
+  recencyWeight: number;
+}
+
+export class RecencyRankedRetriever extends BaseRetriever {
+  static lc_name() {
+    return "RecencyRankedRetriever";
+  }
+
+  lc_namespace = ["langchain", "retrievers", "recency_ranked"];
+  
+  private vectorStore: VectorStoreInterface;
+
+  private k: number;
+
+  private top_k: number;
+
+  private recencyWeight: number;
+
+  constructor(config: RecencyRankedRetrieverConfig) {
+    super();
+    this.vectorStore = config.vectorStore;
+    this.k = config.k;
+    this.top_k = config.top_k ?? config.k;
+    this.recencyWeight = config.recencyWeight;
+  }
+
+  async getRelevantDocuments(query: string): Promise<Document[]> {
```

Comment:
You should override `_getRelevantDocuments` rather than the non-underscored method:

https://js.langchain.com/v0.2/docs/how_to/custom_retriever